### PR TITLE
audit upgrade; direct rule selection; add tests

### PR DESCRIPTION
**Pull request by alagopus**

### CODE REVIEW

**Comment by alagopus** [comment]
If @niallo concurs, I would be okay with moving to MIT license to be in line with the strider packages.

**Comment by knownasilya** [comment]
How are you using this lib?

**Comment by knownasilya** [comment]
Glad to merge this, is it a breaking change?

File: index.js [hunk 1]
```diff
@@ -1,19 +1,85 @@
...
 
-// *rules* - list of detection rules
-// *phase* - which phase to run matches for (prepare, test, deploy, cleanup)
-// *ctx* - Strider worker plugin context
-// *cb* - callback function, takes
-module.exports = function(rules, phase, ctx, cb) {
+/**
+ * Resolve property on context.
+ */
+function resolve(context, property) {
```
Comment (knownasilya) [inline index.js:10]: I'd rename this because of Promise.resolve, it might be confusing

File: index.js [hunk 2]
```diff
@@ -1,19 +1,85 @@
...
+ * Detect and run rule from a set of rules.
+ * @param {object} rules list of detection rules
+ * @param {string} phase which phase to run matches for (prepare, test, deploy, cleanup)
+ * @param {object} ctx Strider worker plugin context
+ * @param {function} cb callback function, takes exit code of rule command
+ */
+function detection(rules, phase, ctx, cb) {
+  const baseDir = ctx.workingDir;
 
+  const work = rules.map((rule) => glob(rule.filename, { cwd: baseDir })
```
Comment (knownasilya) [inline index.js:49]: Why does `map` return a promise here, isn't it an array?

File: index.js [hunk 3]
```diff
@@ -1,19 +1,85 @@
...
-  gumshoe.run(ctx.workingDir, rules, function(err, result) {
-      if (err) return cb(0)
-      var psh = ctx.shellWrap(result[phase])
-      ctx.forkProc(ctx.workingDir, psh.cmd, psh.args, function(code) {
-        return cb(code)
-      })
-  })
+/**
+ * Detect and run rule from a set of rules.
+ * @param {object} rules list of detection rules
```
Comment (knownasilya) [inline index.js:41]: This should be array

**Review by knownasilya — CHANGES_REQUESTED** [review]
Couple small things, otherwise good. Sorry it took so long to see this.

**Review by knownasilya — APPROVED** [review]

File: index.js (hunk 3 above)
Comment (alagopus) [reply index.js:41]: Yes. Sorry for the late response. Keeping it now since already merged.

**Pull request merged**